### PR TITLE
DM-7264: Port to Python 3

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: examples/demoPool.py [hunk 1]
```diff
@@ -35,7 +37,7 @@ def test2(cache, data, *args, **kwargs):
 
 startPool()
 
-dataList = map(float, range(NUM))
+dataList = list(map(float, range(NUM)))
```
Comment (timj) [inline examples/demoPool.py]:
```python
dataList = [float(i) for i in range(NUM)]
```
?

File: python/lsst/ctrl/pool/pool.py [hunk 2]
```diff
@@ -530,7 +533,7 @@ def map(self, context, func, dataList, *args, **kwargs):
         tags = Tags("result", "work")
         num = len(dataList)
         if self.size == 1 or num <= 1:
-            return self._processQueue(context, func, zip(range(num), dataList), *args, **kwargs)
+            return self._processQueue(context, func, list(zip(list(range(num)), dataList)), *args, **kwargs)
```
Comment (timj) [inline python/lsst/ctrl/pool/pool.py]: I don't think the second `list` is needed.

File: python/lsst/ctrl/pool/parallel.py [hunk 3]
```diff
@@ -211,7 +220,7 @@ def preamble(self, walltime=None):
         filename = os.path.join(outputDir, (self.jobName if self.jobName is not None else "slurm") + ".o%j")
         return "\n".join([("#SBATCH --nodes=%d" % self.numNodes) if self.numNodes > 0 else "",
                           ("#SBATCH --ntasks-per-node=%d" % self.numProcsPerNode) if
-                              self.numProcsPerNode > 0 else "",
+                          self.numProcsPerNode > 0 else "",
```
Comment (timj) [inline python/lsst/ctrl/pool/parallel.py]: Does `flake8` like this indenting?

File: examples/demoPool.py (hunk 1 above)
Comment (fred3m) [inline examples/demoPool.py]: Yeah... that's much better

File: python/lsst/ctrl/pool/pool.py (hunk 2 above)
Comment (fred3m) [inline python/lsst/ctrl/pool/pool.py]: You're right

File: python/lsst/ctrl/pool/parallel.py (hunk 3 above)
Comment (fred3m) [inline python/lsst/ctrl/pool/parallel.py]: This was created by autopep8, and flake8 didn't flag it

**Pull request merged**